src/test/java/com/datalogics/pdf/security/LunaHsmManagerTest.java

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright 2016 Datalogics Inc.
+ */
+
+package com.datalogics.pdf.security;
+
+import static org.junit.Assert.assertEquals;
+
+import com.datalogics.pdf.hsm.samples.mock.MockProvider;
+import com.datalogics.pdf.security.HsmManager.ConnectionState;
+
+import mockit.Mock;
+import mockit.MockUp;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import com.safenetinc.luna.LunaException;
+import com.safenetinc.luna.LunaSlotManager;
+import com.safenetinc.luna.provider.LunaProvider;
+
+import java.security.PrivateKey;
+import java.security.Security;
+
+/**
+ * Unit tests for the LunaHsmManager.
+ */
+public class LunaHsmManagerTest {
+    public static final String GOOD_PASSWORD = "good_password";
+    public static final String BAD_PASSWORD = "bad_password";
+    public static final String GOOD_SLOT_NAME = "good_slot_name";
+    public static final String BAD_SLOT_NAME = "bad_slot_name";
+    public static final String KEY_LABEL = "key_label";
+
+    private LunaHsmManager lunaHsmManager;
+
+    @Rule
+    public ExpectedException expected = ExpectedException.none();
+
+    /**
+     * Install mock Luna classes.
+     */
+    @Before
+    public void setUp() {
+        new MockLunaSlotManager();
+        new MockLunaProvider();
+
+        lunaHsmManager = (LunaHsmManager) HsmManagerFactory.newInstance(HsmManagerFactory.LUNA_HSM_TYPE);
+    }
+
+    @Test
+    public void successfulLogin() {
+        lunaHsmManager.hsmLogin(new LunaHsmLoginParameters(GOOD_PASSWORD));
+
+        assertEquals("LunaHsmManager login should be successful", ConnectionState.CONNECTED,
+                     lunaHsmManager.getConnectionState());
+    }
+
+    @Test
+    public void successfulLoginWithSlotName() {
+        lunaHsmManager.hsmLogin(new LunaHsmLoginParameters(GOOD_SLOT_NAME, GOOD_PASSWORD));
+
+        assertEquals("LunaHsmManager login with good slot name should be successful", ConnectionState.CONNECTED,
+                     lunaHsmManager.getConnectionState());
+    }
+
+    @Test
+    public void unsuccessfulLoginThrowsException() {
+        // Expect a IllegalArgumentException to be thrown
+        expected.expect(IllegalArgumentException.class);
+        expected.expectMessage("Error while logging into the Luna HSM");
+
+        lunaHsmManager.hsmLogin(new LunaHsmLoginParameters(BAD_PASSWORD));
+
+        assertEquals("LunaHsmManager login should be unsuccessful", ConnectionState.CONNECTED,
```

Comment:
Is the assertEquals at line 77 really needed as the expected IllegalArgumentException will be thrown at line 75 and this test will pass?